src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/CpsProjectSystemReferenceReader.cs

```diff
@@ -131,5 +131,11 @@ private static async Task<IBuildDependencyProjectReferencesService> GetProjectRe
         {
             throw new NotSupportedException();
         }
+
+        public async Task<IReadOnlyList<(string id, string[] metadata)>> GetItemsAsync(string itemTypeName, params string[] metadataNames)
```

Comment:
Does CPS implement that API? 
I remember it being added to the old project system when we wanted CPM to work.

Reply:
I don't know. There's no code path that calls `GetItemsAsync` currently on `CpsProjectSystemReferenceReader`. It's just needed because it implements the same interface as I need for packages.config (so, software design bug).

I thought it might be better to call the same implementation as the other readers, because maybe it works, rather than throw a `NotImplementedException`.  But I can be persuaded to throw.

Reply:
If it's not implemented, would it crash when GetItems is called?
Unless I misread the code there.

Reply:
same response as https://github.com/NuGet/NuGet.Client/pull/5918#discussion_r1692343413

Reply:
Given my misunderstanding about MSBuildNuGetProject (not) being packages.config only, if I move the `GetItemsAsync` method out of `IProjectSystemReferenceReader` and into `MSBuildNuGetProject`, then CPS won't need to implement it, but CLI restore will (it can noop/throw, since it won't be called in code path).

It's a question of what we think the best design is, as every alternative I can see as feasible has drawbacks.